cmd/kubeadm/app/discovery/file/file.go

```diff
@@ -140,6 +102,54 @@ func ValidateConfigInfo(config *clientcmdapi.Config, clustername string) (*clien
 	return refreshedBaseKubeConfig, nil
 }
 
+// get kubeconfig from config
+func getKubeconfig(config *clientcmdapi.Config, defaultCluster *clientcmdapi.Cluster, clustername string) (*clientcmdapi.Config, error) {
```

Comment:
IMO default ClusterConfiguration should not be passed as a parameter
It is derived from config (that is already a parameter) and never used outside of this function

Reply:
The new function getKubeconfig is extracts the part of create kubeconfig from ValidateConfigInfo. I have added the unit test for getKubeconfig.